tests/OAuth/integration.test.js

```diff
@@ -0,0 +1,104 @@
+const url = require('url')
+const qs = require('querystring')
+
+const tape = require('tape')
+const supertest = require('supertest')
+
+const server = require('../../src/server')
+
+const Client = require('../../src/models/auth/Client.js')
+const User = require('../../src/models/User.js')
+const AuthorizationCode = require('../../src/models/auth/AuthorizationCode.js')
+const Token = require('../../src/models/auth/Token.js')
+
+const { client } = require('../fixtures/auth/clients.json')
+const { user } = require('../fixtures/users.json')
+
+tape('emptying db.', t => {
+  Promise.resolve()
+  .then(() => User.remove({}))
+  .then(() => Client.remove({}))
+  .then(() => AuthorizationCode.remove({}))
+  .then(() => Token.remove({}))
+  .then(() => t.end())
+  .catch(err => t.end(err))
+})
+
+// Tests for: GET /oauth/authorize
+// should return custom page (handlebars?) showing the app, the owner, the requested permissions (if we implement this)
+// tape('GET /oauth/authorize, should return form page', t => {
+//   supertest(server)
+//     .get('/oauth/authorize')
+//     .query({
+//       client_id: 'testClientId',
+//       redirect_uri: 'testRedirect'
+//     })
+//     .expect(200)
+//     .expect('Content-Type', /html/)
+//     .end((err, res) => {
+//       t.error(err)
+//       t.end()
+//     })
+// })
+
+// the first test creates the authCode, the second test uses it
+let authCode
+
+// Tests for: POST /oauth/authorize
+tape('POST /oauth/authorize should successfully redirect and create authorization code', t => {
+  Promise.resolve()
+  // add valid user, client
+  .then(() => User.create(user))
+  .then(() => Client.create(client))
+  // do the test
+  .then(() => {
+    const randomState = 'randomString'
+    supertest(server)
```

Comment:
we can flatten this one level, ie `return supertest(...`

You can put random state one level out also